x-pack/filebeat/docs/inputs/input-http-endpoint.asciidoc

```diff
@@ -11,9 +11,31 @@
 
 beta[]
 
-Use the `http_endpoint` input to create a HTTP listener that can receive incoming HTTP POST requests.
-
-This input can for example be used to receive incoming webhooks from a third-party application or service.
+The HTTP Endpoint input initializes a listening HTTP server that collects
+incoming HTTP POST requests containing a JSON body. The body must be either an
+object or an array of objects. Any other data types will result in an HTTP 400
+(Bad Request) response. For arrays, one document is created for each object in
+the array.
+
+gzip encoded request bodies are supported if a `Content-Encoding: gzip` header
+is sent with the request.
```

Comment:
Not mentioning x-gzip I think is the correct approach; we accept it, but don't advertise the fact.